src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -75,24 +77,26 @@ bool CLinearRidgeRegression::train_machine(CFeatures* data)
 	for (int32_t i=0; i<num_feat; i++)
 		kernel_matrix.matrix[i+i*num_feat]+=m_tau;
 
+	// Get vectors and matrices in eigen form
+	Map<MatrixXd> eigen_kernel_matrix(kernel_matrix.matrix, num_feat,num_feat);
+	Map<VectorXd> eigen_y(y.vector, num_feat);
+
 	for (int32_t i=0; i<num_vec; i++)
```

Comment:
No need for a loop here. You can compute the covariance matrix as a matrix-matrix product. Use linalg
See e.g. http://web4.cs.ucl.ac.uk/staff/D.Barber/pmwiki/pmwiki.php?n=Brml.Software